### ISSUE TITLE
Build is broken due to ethereumjs-vm

### ISSUE THREAD

**Issue by axic** [ethereum/browser-solidity#822]
Since currently remix points to `master` of ethereumjs-vm it is vulnerable to changes in there. Just today Byzantium changes were merged and that broke the build, at least for one reason: it contains an Emscripten asm.js version of the Rust pairing library and that results in:
```
TypeError: str.charCodeAt is not a function. (In 'str.charCodeAt(i)', 'str.charCodeAt' is undefined)
```

**Comment by jwasinger** [comment]
Looking into this.

**Comment by holgerd77** [comment]
Just a bit more context to this error:

When following the [installation instructions](https://github.com/ethereum/browser-solidity#installation) from the library and start a local developer server with ``npm start`` and opening http://127.0.0.1:8080/ the screen stays blank and the following error occurs in the developer console:

```
Uncaught TypeError: str.charCodeAt is not a function
    at lengthBytesUTF8 (app.js:124111)
    at intArrayFromString (app.js:124111)
    at Object.callMain (app.js:124127)
    at doRun (app.js:124127)
    at run (app.js:124127)
    at Object.<anonymous> (app.js:124127)
    at Object.804._process (app.js:124133)
    at s (app.js:1)
    at app.js:1
    at Object.803../lib/index.asm.js (app.js:124106)
```

**Comment by holgerd77** [comment]
Some more investigation on the error:

1) This also occurs when build directly in ``rustbn.js`` with ``browserify -e index.js -o exp/out.js`` and then ``out.js`` included in the ``exp/index.html`` file.

2) When searching for ``var u=str.charCodeAt(i);if(u>=55296&&u<=57343)u=65536`` in the generated ``app.js`` in ``browser-solidity`` and adding ``if (!str.charCodeAt) return 0;`` directly before the two result lines found, the error disappears/changes to:

```
06-ecadd.js:7 Uncaught TypeError: bn128Module.cwrap is not a function
    at Object.<anonymous> (06-ecadd.js:7)
    at Object.601.../constants.js (06-ecadd.js:9)
    at s (_prelude.js:1)
    at _prelude.js:1
    at Object.592../precompiled/01-ecrecover.js (index.js:15)
    at s (_prelude.js:1)
    at _prelude.js:1
    at Object.587../lib/index.js (index.js:1)
    at s (_prelude.js:1)
    at _prelude.js:1
```

**Comment by holgerd77** [comment]
The above error is just happening because when bridging ``lengthBytesUTF8`` by returning ``0``, no code is added, so ``bn128`` is an empty object.

The error lies somewhere in the ``index.asm.js`` include in the ``rustbn.js`` library. Maybe this can be wrapped somehow or included in some other way, experimented a bit on this but no success yet.

**Comment by holgerd77** [comment]
Maybe experimenting with https://kripken.github.io/emscripten-site/docs/api_reference/module.html could also be helpful.

**Comment by holgerd77** [comment]
Also maybe helpful: https://github.com/kripken/emscripten/issues/4032

**Comment by holgerd77** [comment]
@axic Fix merged, this can be closed now.

**Comment by axic** [comment]
Added a new issue for upgrading once released: #828
